### DISCUSSION TITLE
How to add a camera shot from 3d view to the sequencer?

### DISCUSSION THREAD

**Discussion by tin2tin** [tin2tin/Pallaidium#52] [Cookbook - share your genAI recipes here]
Trim strip - the trimmed strip will now be processed instead of the source strip, when doing vid2vid. 
OpenPose now can input OpenPose rig from the Blender 3D space (or transfer the pose from a photo).

https://github.com/tin2tin/Pallaidium/assets/1322593/5ae5d5bd-31e8-490d-800f-e5186a5670be

- Open an OpenPose rig in the 3D View. In the video I use this one: https://impactframes.gumroad.com/l/fxnyez
- Make a Linked copy of the Scene:  
![image](https://github.com/tin2tin/Pallaidium/assets/1322593/49d37db0-6c69-480a-bce5-01ab07abf4d8)
- Swich back to the Scene with the rig.
- Open the Video Sequence Editor.
- Add a scene strip - the Linked Copy version of the scene. 
- When doing changes in the 3D View, move the cursor to the Sequencer and hit Ctrl+R to update the Sequencer Preview. 
- Select Image > OpenPose in the Pallaidium preferences.
- Select the Scene strip.
- Select OpenPose Rig Image as Input.
- Type in a prompt, and hit Generate.

ControlNet guess mode was faulty, and works better without it. 

Example of 3d to ControlNet:

https://github.com/tin2tin/Pallaidium/assets/1322593/7fb718e0-76a6-4384-ab0d-a1f2aefb6f94

- Make a scene in the 3D View out of simple objects. 
- Make a Linked Copy of the Scene:
![image](https://github.com/tin2tin/Pallaidium/assets/1322593/49d37db0-6c69-480a-bce5-01ab07abf4d8)
- Swich back to the original Scene.
- Open the Video Sequence Editor.
- Add a Scene Strip - the Linked Copy version of the scene. 
- When doing changes in the 3D View, move the cursor to the Sequencer and hit Ctrl+R to update the Sequencer Preview. 
- Select Image > ControlNet in the Pallaidium add-on preferences.
- Select the Scene strip.
- Select  Input: Strips
- Type in a prompt, and hit Generate.

**Comment by slrgt** [comment]
amazing thanks! 

It would be convenient if you used the Releases feature so we can get updates from new releases

**Reply by tin2tin** [reply]
As long as I'm updating the script several times a day, I rather not spend time on uploading zipped release files. This is still the very early days of this add-on.

**Comment by djooodj** [comment]
Thank you for this! I am trying the 3d to controlNet, following the above instructions. All works as expected except I'm unable to get it to respect my camera position. I only have one camera, and it is set to active. I've control R clicked in the preview window to update linked scene to the latest camera position. Thoughts?

**Comment by tin2tin** [comment]
Maybe the Linked Copy doesn't update changes in the 3D scene anymore, so try to make a fresh linked scene and point the scene strip to that one.

**Reply by djooodj** [reply]
I'm now having trouble getting it to work at all, and I'm getting this error:
Python: Traceback (most recent call last):
  File "C:\Users\USER PROFILE NAME\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\Pallaidium-main\__init__.py", line 3195, in execute
    temp_strip = strip = get_render_strip(
  File "C:\Users\USER PROFILE NAME\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\Pallaidium-main\__init__.py", line 1272, in get_render_strip
    bpy.ops.sequencer.movie_strip_add(
  File "C:\Program Files\Blender Foundation\Blender 3.6\3.6\scripts\modules\bpy\ops.py", line 113, in __call__
    ret = _op_call(self.idname_py(), None, kw)
RuntimeError: Error: File 'C:\Users\USER PROFILE NAME\AppData\Roaming\Blender Foundation\Blender\3.6\datafiles/Rendered_Strips_2023-09-28\Scene.006_rendered.mp4' could not be loaded

**Comment by tin2tin** [comment]
Try the update. I hope I fixed it.

**Reply by djooodj** [reply]
This update fixed it, thank you!

**Reply by djooodj** [reply]
I'm not sure if this is related, but since updating, I'm having trouble using the Illusion model, see attached image. It looks like it's trying to use ControlNet? I could be totally off on this, but results no longer seem to incorporate my input strip. 
![pallaidium_console_screengrab_illlusion_issue](https://github.com/tin2tin/Pallaidium/assets/146207878/9987f19d-2955-4f92-be01-f758b37c0c3f)

**Comment by tin2tin** [comment]
Yeah, I broke something. I noticed myself right now. But I do not have time to fix it atm.

**Comment by djooodj** [comment]
oh no worries, I appreciate your work so much. Looking forward to the
update whenever it comes.

**Reply by tin2tin** [reply]
Made a quick update, but I do not have time to check if everything is working as intended right now, but at least should that error be gone now.

**Reply by djooodj** [reply]
Hi there, I am still seeing that indicating the Illusion model is still calling ControlNet, and illusion is not working correctly. I just downloaded the latest zip (including added storyboard styles). I assume this update would be included in this package?

**Reply by tin2tin** [reply]
If there is a bug, you should properly do a report and include steps to reproduce and error mesages etc. 
I just tried the Illusion model here, and it seems to be working as intended(no errors).

**Reply by djooodj** [reply]
thanks, I'm a little new to github. Do you want me to enter it as an "issue"? Or just respond to this chat with more detail?

**Reply by tin2tin** [reply]
Yes, open an issue.
